src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanicsKernel.hpp

```diff
@@ -89,15 +89,15 @@ class Multiphase :
               SUBREGION_TYPE const & elementSubRegion,
               FE_TYPE const & finiteElementSpace,
               CONSTITUTIVE_TYPE & inputConstitutiveType,
-              arrayView1d< globalIndex const > const & inputDispDofNumber,
-              string const & inputFlowDofKey,
+              arrayView1d< globalIndex const > const inputDispDofNumber,
+              string const inputFlowDofKey,
               globalIndex const rankOffset,
               real64 const (&inputGravityVector)[3],
               localIndex const numComponents,
               localIndex const numPhases,
-              arrayView1d< string const > const fluidModelNames,
-              CRSMatrixView< real64, globalIndex const > const & inputMatrix,
-              arrayView1d< real64 > const & inputRhs ):
+              string const fluidModelKey,
+              CRSMatrixView< real64, globalIndex const > const inputMatrix,
+              arrayView1d< real64 > const inputRhs ):
```

Comment:
I have two questions:
- What was wrong with passing the `inputMatrix` and `inputRhs` by reference?
- After this PR, should we start removing references where they are still present (for instance, in `CompositionalMultiphaseBaseKernels` and possibly in the wells?

Reply:
In this particular case, probably nothing wrong. In fact, I'd say the change to kernel constructor signature is not that important, it's just there to match the change to `MultiphaseKernelFactory` alias below. @rrsettgast found a subtle bug: instance of a factory class keeps the values given to it in a tuple for some amount of time before the instance of a kernel is created from those values; if you happen to pass a temporary to the factory (such as a `string` created from a `const char *`, or a `CRSMatrixView` created from a `CRSMatrix` on the fly), it'll store a reference to that temporary which becomes dangling as soon as the factory constructor is finished executing, so when we later call `.createKernel()` it passes dangling references to the kernel constructor. This hasn't happened specifically to `inputMatrix` and `inputRhs` so far because those temporary view object generally have lifetimes for the entire duration of assembly, but as a safety policy we shouldn't be passing references to factories unless they refer to really persistent objects (such as various mesh object managers or other high-level components).